api/views.py

```diff
@@ -342,9 +446,18 @@ def get(self, request, proteins=None, segments=None):
             
             if segments is not None:
                 segment_list = segments.split(",")
-                ss = ProteinSegment.objects.filter(slug__in=segment_list, partial=False)
-            else:
-                ss = ProteinSegment.objects.filter(partial=False)
+                generic_list = []
+                TM_list = []
+                for s in segment_list:
+                    # get genric numbering object for numbers and segments for TMs
+                    if "x" in s:
```

Comment:
This is very specific to the GPCR numbering scheme. Would be better to check if the "segment" is in the segment table (just fetch a list of all segments in the beginning, and then check everything against that list, that is faster than doing a DB lookup every time), and if not, treat it as a generic number.